// Copyright 2021 The Cross-Media Measurement Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// The protobufs here are only used for testing purpose.
syntax = "proto3";

package wfa_virtual_people.test;

message TestProto {
  optional TestProtoA a = 1;

  repeated int32 int32_values = 2;

  repeated TestProtoA repeated_proto_a = 3;
}

message TestProtoA {
  optional TestProtoB b = 1;
}

message TestProtoB {
  enum TestEnum {
    INVALID = 0;
    TEST_ENUM_1 = 1;
    TEST_ENUM_2 = 2;
    TEST_ENUM_3 = 3;
  }
  optional int32 int32_value = 1;
  optional int64 int64_value = 2;
  optional uint32 uint32_value = 3;
  optional uint64 uint64_value = 4;
  optional float float_value = 5;
  optional double double_value = 6;
  optional bool bool_value = 7;
  optional TestEnum enum_value = 8;
  optional string string_value = 9;

  repeated int32 int32_values = 10;
  repeated int64 int64_values = 11;
  repeated uint32 uint32_values = 12;
  repeated uint64 uint64_values = 13;
  repeated float float_values = 14;
  repeated double double_values = 15;
  repeated bool bool_values = 16;
  repeated TestEnum enum_values = 17;
  repeated string string_values = 18;
}
